src/tools/dev/scripts/bv_support/helper_funcs.sh

```diff
@@ -1,5 +1,48 @@
 export LOG_FILE=${LOG_FILE:-"${0##*/}_log"}
 
+# *************************************************************************** #
+# Purpose: Flexible comparison function for version strings                   #
+#                                                                             #
+#   - Converts version string "4.101.3" to bash array (4 101 3)               #
+#   - Appends zeros to operand with fewer array members (4)==>(4 000 0)       #
+#   - Ensures appended zeros have same length as counterparts                 #
+#   - Forms integer values from arrays (4 101 3)==>41013                      #
+#   - Compares integers using specified operator                              #
+#   - Sets status using test operator                                         #
+#                                                                             #
+# *************************************************************************** #
+function compare_version_strings
+{   
+    # default op is -lt and separator char is .
+    op=${3:-'-lt'}
+    sep=${4:-'.'}
+
+    # create array variables of digits from version string
+    vldigitarr=($(echo $1 | tr $sep ' '))
+    vrdigitarr=($(echo $2 | tr $sep ' '))
+
+    # append strings of zeros of equal length for missing digits
+    # "5"==>"0", "10"==>"00", "101"==>"000"
+    i=0
+    while [[ ${#vldigitarr[@]} -lt ${#vrdigitarr[@]} ]]; do
+        zeros=$(echo ${vrdigitarr[$i]} | tr '1234567890' '0000000000')
```

Comment:
Problem here comparing a strings like `4.9.0` (bash array `(4 9 0)`) and `4.10.0` (bash array `(4 10 0)`) becuase a zero needs to get appended to the `9` in the first array for `(4 90 0)`.

Reply:
Whoops...not *appended* but *pre-pended*

Reply:
So, we wind up comparing `4100` and `4090`

Reply:
Stuffing this ref here so I don't forget it, https://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo